app/assets/javascripts/common/reports.js

```diff
@@ -410,21 +410,29 @@ Reports = function () {
         const mostRecentPeriod = cardNode.getAttribute("data-period");
         const totalPatientsNode = cardNode.querySelector("[data-total-patients]");
         const registrationsPeriodEndNode = cardNode.querySelector("[data-registrations-period-end]");
+        const registrationsEndDate = registrationsPeriodEndNode.getAttribute("data-registrations-period-end");
         const monthlyRegistrationsNode = cardNode.querySelector("[data-monthly-registrations]");
         const registrationsMonthEndNode = cardNode.querySelector("[data-registrations-month-end]");
+        const totalRegistrationDataPoints = Object.values(data.cumulativeRegistrations).length - 1;
         let label = null;
+        let dataPointIndex = null;
         if (tooltip.dataPoints) {
           label = tooltip.dataPoints[0].label;
+          dataPointIndex = tooltip.dataPoints[0]["index"];
         } else {
           label = mostRecentPeriod;
         }
         const period = data.periodInfo[label];
         const cumulativeRegistrations = data.cumulativeRegistrations[label];
         const monthlyRegistrations = data.monthlyRegistrations[label];
 
+        const setRegistrationEndDate = dataPointIndex == totalRegistrationDataPoints || tooltip.dataPoints == undefined
+          ? registrationsEndDate
+          : period.bp_control_end_date;
```

Comment:
When is `tooltip.dataPoints` undefined? Is it in the case of no data: Do we separately need to handle that case? Trying to figure out if this can be simplified.

Reply:
Thanks! I removed this change as part of the LTFU revert :)